src/web_video_server.cpp

```diff
@@ -201,13 +196,13 @@ bool WebVideoServer::handle_stream(
       }
       if (!did_find_compressed_topic) {
         RCLCPP_WARN(
-          node_->get_logger(),
+          get_logger(),
           "Could not find compressed image topic for %s, falling back to mjpeg", topic.c_str());
         type = "mjpeg";
       }
     }
     std::shared_ptr<ImageStreamer> streamer = stream_types_[type]->create_streamer(
-      request, connection, node_);
+      request, connection, shared_from_this());
```

Comment:
This is what I'm worried about a little. `shared_from_this()` will throw `std::bad_weak_ptr` if there is no `shared_ptr` created yet, so there is a possibility of a race condition after call to `server_->run();`. I have not managed to trigger that race condition and we are catching all exceptions in `handle_request()` so we should probably be good.

Reply:
The WVS is constructed as a shared pointer and `shared_from_this()` isn't called until a callback handler is invoked by a client connection.  By this time, the server should be up-and-running.  Your point about catching all exceptions in the request handler just provides additional assurance against any runtime application crashes.